src/API.jl

```diff
@@ -20,14 +20,21 @@ include("generate.jl")
 
 function check_package_name(x::AbstractString)
     if !(occursin(Pkg.REPLMode.name_re, x))
-         pkgerror("$x is not a valid packagename")
+        islikelyurl(x) = any(occursin.(['\\','/'], x))
+        message = "$x is not a valid packagename."
+        if islikelyurl(x)
+            message *= " Did you mean url= or path= \"$x\"?"
+        end
+        pkgerror(message)
     end
-    return PackageSpec(x)
+    return Package(x)
```

Comment:
As described in the comments, `API.Package` is only meant as the API constructor. So this should be `PackageSpec(;name=x)`